### ISSUE TITLE
GSConnect keeps losing pairing to my V2026 phone when it is also paired to an Android x86 virtual machine

### ISSUE THREAD

**Issue by KSXGitHub** [GSConnect/gnome-shell-extension-gsconnect#1696]
### Describe the bug

It used to work fine when I only have KDE Connect installed on my phone. But after I install KDE Connect to my Android x86 (virt-manager) and pair it with GSConnect on the host computer, the host computer repeatedly loses pairing with my phone.

### GSConnect version

55

### Installed from

GNOME Extensions website

### GNOME Shell version

44.4 (downgraded from 45 because most of my extensions, including GSConnect aren't supported)

### Linux distribution/release

Arch Linux

### Paired device(s)

Vivo 2026

### KDE Connect app version

1.29.0

**Comment by ferdnyc** [comment]
@KSXGitHub 

Is the Android emulator instance set up as a "clone" of your physical phone, in any way? Would they share any of the same hardware parameters (device ID, IP address, anything like that) or user data? I'm just wondering if perhaps GSConnect might be seeing them as the **same** device, and therefore interpreting a connection from the VM as a reconnection from your phone, instead.

The other possibility is that GSConnect's service is simply **crashing** when one or both of the devices communicates with it, leading to a reset of all the active connections. If that's the case, there should be some indication in the debug logs of what's going wrong. 

It would be helpful if you could activate Generate Debug Log, then let it keep capturing until the next time your phone gets disconnected and post the results. (There will most likely be a LOT of sensitive data in the log, so be careful about how much you share publicly.)

But in particular, if you have the SFTP plugin enabled for any paired Android device, I'd recommend switching it off from the Linux end (separately for every paired device), then see if that stops the disconnections. 

SFTP in the current releases can be a bit buggy, in ways that sometimes crash GSConnect completely every 5 minutes or so whenever it's enabled.

**Comment by ferdnyc** [comment]
> Generate Debug Log

Sorry, I meant [Generate Support Log](https://github.com/GSConnect/gnome-shell-extension-gsconnect/wiki/Help#generate-support-log). That link has step-by-step instructions on how to do that.

See https://github.com/GSConnect/gnome-shell-extension-gsconnect/issues/1637#issuecomment-1585935018 for information on disabling the SFTP plugin.

**Comment by KSXGitHub** [comment]
> Is the Android emulator instance set up as a "clone" of your physical phone, in any way?

No. The Android virtual machine is a completely unrelated x86 iso.

> I'm just wondering if perhaps GSConnect might be seeing them as the **same** device, and therefore interpreting a connection from the VM as a reconnection from your phone, instead.

I also suspect some IP conflicts to be the reason, not between the phone and the virtual machine, but between the computer host and the virtual machine. I saw KDE Connect on my phone frequently loses connection. Instead of recognizing the name of my desktop computer, it recognizes the name of the virtual machine. Only after I turn off the virtual machine, unpair and pair again, did it start to work properly.

> The other possibility is that GSConnect's service is simply crashing when one or both of the devices communicates with it, leading to a reset of all the active connections. 

Could be the case too. I also observed that, sometimes when both phone and virtual machine are active, clicking "Mobile Settings" does respond at all. And the KDE Connect on phone does not connect with my desktop computer.

> But in particular, if you have the SFTP plugin enabled for any paired Android device, I'd recommend switching it off from the Linux end (separately for every paired device), then see if that stops the disconnections.

Will do.

**Comment by KSXGitHub** [comment]
**UPDATE:** I just booted up my virtual machine, make KDE Connect inside it request pairing, then accept pairing in the GSConnect's notification.

The result:
* It pairs with the virtual machine.
* It unpairs my phone.

[Support Log](https://gist.githubusercontent.com/KSXGitHub/d0db4585352efcfd85c263862858bdbe/raw/6d2b1c53d572ae753c84a85b37c4f15ae7faaed0/1.%2520accepting%2520pairing%2520request%2520from%2520the%2520virtual%2520machine).

**Comment by KSXGitHub** [comment]
**UPDATE:** When I use GSConnect UI to pair with my phone again, my phone is paired again, and both my phone and my virtual machine appeared as "Connected" in GSConnect UI. **However,** only my phone is actually connected, the virtual machine does not see the connection at all.

[Support Log](https://gist.githubusercontent.com/KSXGitHub/d0db4585352efcfd85c263862858bdbe/raw/6d2b1c53d572ae753c84a85b37c4f15ae7faaed0/2.%2520accepting%2520pairing%2520request%2520from%2520the%2520phone).
